dace/transformation/passes/simplification/control_flow_raising.py

```diff
@@ -167,32 +201,103 @@ def _lift_conditionals(self, sdfg: SDFG) -> int:
                         graph.remove_nodes_from(branch_nodes)
 
                     # Connect to the end of the branch / what happens after.
-                    if dummy_exit is None or merge_block is not dummy_exit:
+                    if (dummy_exit is None or merge_block is not dummy_exit) and merge_block is not None:
                         graph.add_edge(conditional, merge_block, InterstateEdge())
             if dummy_exit is not None:
                 region.remove_node(dummy_exit)
 
         n_cond_regions_post = len([x for x in sdfg.all_control_flow_blocks() if isinstance(x, ConditionalBlock)])
         lifted = n_cond_regions_post - n_cond_regions_pre
         if lifted:
+            sdfg.reset_cfg_list()
             sdfg.root_sdfg.using_explicit_control_flow = True
         return lifted
 
+    def _lift_unstructured(self, sdfg: SDFG) -> int:
```

Comment:
I'm not sure that we have enough tests for this. e.g., two loops with one jumping to the middle of the next (sibling), branch jumping to the branch contents of another if in the else clause (tree-depth change), etc.

Another key thing to test is that the Unstructured CF region contains all the nodes that also may be the targets of the jumps, rather than just SESE bounds. However, I am not 100% certain that we must do this goto target boundaries; we do it in schedule trees.

Reply:
As a result, this PR now closes https://github.com/spcl/dace/issues/2062